services/httpd/handler.go

```diff
@@ -417,6 +419,11 @@ func (h *Handler) serveWriteJSON(w http.ResponseWriter, r *http.Request, body []
 		return
 	}
 
+	if !h.JSONWriteEnabled {
```

Comment:
I think this could be done at the very beginning of the func and avoid spending time decoding the JSON payload.